core/src/main/java/io/undertow/server/DefaultByteBufferPool.java

```diff
@@ -343,7 +344,7 @@ protected void finalize() throws Throwable {
     // class can be called by a different thread than the one that initialized the data.
     private static class ThreadLocalCache {
 
-        Map<Thread, ThreadLocalData> localsByThread = new HashMap<>();
+        Map<Thread, ThreadLocalData> localsByThread = Collections.synchronizedMap(new WeakHashMap<>());
```

Comment:
how about using a concurrent hashmap?

Reply:
Good idea @He-Pin ! I'll swap it by ConcurrentHashMap. Thanks!

Reply:
Oh, I just see it now. We don't have a WeakConcurrentHashMap unfortunately... so, we would miss the functionality for WeakReferences being automatically removed. I'll have to stick with the synchronized wrapper for now. Anyway, thanks for the feedback :-)